src/System.Reflection.Metadata.ILDasm/src/ILConstant.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ILDasmLibrary
+{
+    public struct ILConstant
+    {
+        private Readers _readers;
+        private Constant _constant;
+        private ConstantTypeCode _typeCode;
+        private bool _isTypeInitialized;
+        private object _value;
+
+        internal static ILConstant Create(Constant constant, ref Readers readers)
+        {
+            ILConstant ilConstant = new ILConstant();
+            ilConstant._constant = constant;
+            ilConstant._readers = readers;
+            ilConstant._isTypeInitialized = false;
+            return ilConstant;
+        }
+
+        public ConstantTypeCode TypeCode
+        {
+            get
+            {
+                if (!_isTypeInitialized)
+                {
+                    _isTypeInitialized = true;
+                    _typeCode = _constant.TypeCode;
+                }
+                return _typeCode;
+            }
+        }
+
+        public object Value
+        {
+            get
+            {
+                if(_value == null)
+                {
+                    _value = GetValue();
+                }
+                return _value;
+            }
+        }
+
+        public string GetValueString()
+        {
+            BlobReader reader = _readers.MdReader.GetBlobReader(_constant.Value);
+            object value = Value;
+            switch (TypeCode)
+            {
+                case ConstantTypeCode.Byte:
+                    return string.Format("uint8(0x{0})", ((byte)value).ToString("X2"));
```

Comment:
You should do Format("uint8(0x{0:X2})", (byte)value);
This will do less allocations.